File: solutions/codechef/PETSTORE/PETSTORE_3.c
// QUESTION URL: https://www.codechef.com/problems/PETSTORE
// STATUS: accepted
// TIME: 0.02
// MEMORY: 5.3M

#include <stdio.h>

int main(void) {
    int t, n, a, flag, b[100];
    scanf("%d", &t);
    
    while(t--){
        scanf("%d", &n);
        flag = 1;
        
        for (int i=0; i<100; i++) b[i] = 0;
        
        for (int i=0; i<n; i++){
            scanf("%d", &a);
            b[a]++;
        }
        
        for (int i=0; i<100; i++){
            if (b[i] > 0 && b[i] % 2 != 0){
                flag = 0;
                break;
            }
        }
        
        if (flag) printf("YES\n");
        else printf("NO\n");
        
    }
	return 0;
}

File: solutions/codechef/PETSTORE/PETSTORE_3.py
# DATE: 29/12/2022, 07:18:04
# PROBLEM NAME: Pet Store
# PROBLEM URL: https://www.codechef.com/problems/PETSTORE
# PROBLEM DIFFICULTY RATTING: 1126
# STATUS: accepted
# TIME: 0.04
# MEMORY: 10.5M

for _ in range(int(input())):
    n = int(input())
    a = input()
    a_set = set(a)

    for value in a_set:
        if value != " " and a.count(value) % 2 != 0:
            print("NO")
            break
    else:
        print("YES")



